clients/registry-amqp/src/main/java/org/eclipse/hono/client/registry/amqp/ProtonBasedTenantClient.java

```diff
@@ -59,29 +61,37 @@
 /**
  * A vertx-proton based client of Hono's Tenant service.
  *
+ * If a response cache has been provided, a notification receiver can be used to receive notifications about changes in
+ * tenants from Hono's Device Registry. The notifications are used to invalidate the corresponding entries in the
+ * response cache.
  */
 public final class ProtonBasedTenantClient extends AbstractRequestResponseServiceClient<TenantObject, TenantResult<TenantObject>> implements TenantClient {
 
     private static final Logger LOG = LoggerFactory.getLogger(ProtonBasedTenantClient.class);
     private static final StringTag TAG_SUBJECT_DN = new StringTag("subject_dn");
+    private static final String ATTRIBUTE_KEY_TENANT_ID = "tenant-id";
     private final Map<Object, Future<TenantResult<TenantObject>>> pendingRequests = new HashMap<>();
+    private final NotificationReceiver notificationReceiver;
 
     /**
      * Creates a new client for a connection.
      *
      * @param connection The connection to the service.
      * @param samplerFactory The factory for creating samplers for tracing AMQP messages being sent.
+     * @param notificationReceiver The client to receive notifications from the device registry.
      * @param responseCache The cache to use for service responses or {@code null} if responses should not be cached.
      * @throws NullPointerException if any of the parameters other than the response cache are {@code null}.
      */
     public ProtonBasedTenantClient(
             final HonoConnection connection,
             final SendMessageSampler.Factory samplerFactory,
+            final NotificationReceiver notificationReceiver,
```

Comment:
it feels a little inconvenient that you always need to specify a receiver. Can't we overload the constructor and simply assume a Noop receiver by default?

Reply:
I can imagine running the adapters with a custom registry that does not send notifications at all ...

Reply:
I do not quite see why the response cache parameter should be `null` in that case. Caching the results returned from the registry still makes sense, doesn't it? In fact, that is what we are already doing ...

Reply:
> In fact, that is what we are already doing ...

My understanding was that we add the feature to stop doing that :smile: But you are right, we should not couple the caching to the notifications on that level. I added the parameter.